File: exercises/D308/solution.cpp
#include <iostream> 
using namespace std;
int main() {
    string s, t;
    cin >> s;
    cin >> t;
    int len1 = s.length();
    int len2 = t.length();
    int a = 0, b = 0;
    for (int i = 0; i <= len1 - len2; i++) {
        if (s.substr(i, len2) == t) {
            a++;
        }
    }
    for (int i = 0; i <= len1 - len2; i++) {
        if (s.substr(i, len2) == t) {
            b++;
            i += len2 - 1;
        }
    }
    cout << a << endl;
    cout << b << endl;
}

File: exercises/D308/solution.py
s = input()
t = input()
a = 0
b = 0
len1 = len(s)
len2 = len(t)
for i in range(len1 - len2 + 1):
    if  s[i: i + len2] == t:
        a += 1
i = 0
while i <= len1 - len2:
    if  s[i: i + len2] == t:
        b += 1
        i += len2 - 1
    i += 1
print(a)
print(b)
